example_code/shmem_team_context.c

```diff
@@ -1,107 +1,80 @@
 #include <shmem.h>
 #include <stdio.h>
 
-int isum, ival;
 
-int my_ctx_translate_pe(shmem_ctx_t src_ctx, int src_pe, shmem_ctx_t dest_ctx) {
-  if (src_ctx == SHMEM_CTX_INVALID) {
-    return -1;
-  }
-  if (dest_ctx == SHMEM_CTX_INVALID) {
-    return -1;
-  }
-
-  shmem_team_t src_team, dest_team;
-  shmem_ctx_get_team(src_ctx, &src_team);
-  shmem_ctx_get_team(dest_ctx, &dest_team);
-  return shmem_team_translate_pe(src_team, src_pe, dest_pe);
-}
-
-shmem_ctx_t my_team_create_ctx(shmem_team_t team) {
-  if (team == SHMEM_TEAM_INVALID) {
-    return SHMEM_CTX_INVALID;
-  }
+int main(void)
+{
+  static int            sum = 0, val_2, val_3;
+  shmem_team_t          team_2, team_3;
+  shmem_ctx_t           ctx_2, ctx_3;
+  shmem_team_config_t   conf;
 
-  shmem_ctx_t ctx;
-  if (shmem_team_create_ctx(team, 0, &ctx) != 0) {
-    fprintf(stderr, "Failed to create context for PE team\n");
-    return SHMEM_CTX_INVALID;
-  }
-  return ctx;
-}
-
-void my_send_to_neighbor(shmem_ctx_t ctx, int *val) {
-  if (ctx == SHMEM_CTX_INVALID) {
-    fprintf(stderr, "Send to neighbor fail due to invalid context\n");
-    return;
-  }
+  shmem_init();
 
-  shmem_team_t team;
-  shmem_ctx_get_team(ctx, &team);
-  int team_mype = shmem_team_my_pe(team);
-  int team_npes = shmem_team_n_pes(team);
-  int rpe = (team_mype + 1) % team_npes;
+  int npes  = shmem_n_pes();
+  int my_pe = shmem_my_pe();
+  conf.num_contexts = 1;
+  long cmask = SHMEM_TEAM_NUM_CONTEXTS;
 
-  // put my pe number in the buffer on my right hand neighbor
-  shmem_ctx_int_put(ctx, val, &pe, 1, rpe);
-}
+  /* Create team_2 with PEs numbered 0, 2, 4, ... */
+  shmem_team_split_strided(SHMEM_TEAM_WORLD, 0, 2, (npes + 1) / 2, &conf, cmask, &team_2);
 
-int main() {
-  shmem_init();
+  /* Create team_3 with PEs numbered 0, 3, 6, ... */
+  shmem_team_split_strided(SHMEM_TEAM_WORLD, 0, 3, (npes + 2) / 3, &conf, cmask, &team_3);
 
-  int npes = shmem_n_pes();
-  isum = 0;
+  /* Create a context on team_2. */
+  int ret = shmem_team_create_ctx(team_2, 0, &ctx_2);
 
-  shmem_team_t team_2s, team_3s;
-  shmem_ctx_t ctx_2s, ctx_3s;
-  shmem_team_config_t conf = {.num_contexts = 1};
-  long cmask = SHMEM_TEAM_NUM_CONTEXTS;
+  /* Create a context on team_3. */
+  ret = shmem_team_create_ctx(team_3, 0, &ctx_3);
```

Comment:
Don't we need to check whether context creation was successful?

Reply:
I think so, thanks!